#**Exploratory Data Analysis**
This project involves a thorough analysis of the selected data to identify patterns, trends, and relevant insights.

**Description:** Choose a dataset of interest and explore its characteristics. Clean and prepare the data, then use techniques of visualization, descriptive statistics, and interactive exploration to identify valuable insights. Highlight noteworthy patterns, trends, or anomalies present in the data.

#**About the Dataset**
The dataset for this project covers information on salaries for various positions within the field of Data Science. It provides a detailed view of remuneration across different roles, such as Data Scientist, Machine Learning Engineer, Data Analyst, among others.

**Dataset available at:** https://www.kaggle.com/datasets/arnabchaki/data-science-salaries-2023

#**Data Cleaning and Preparation**
I like to start by analyzing the original dataset in search of interesting data.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('ds_salaries.csv')

df.head(len(df))

work_year experience_level employment_type                 job_title  \
0          2023               SE              FT  Principal Data Scientist   
1          2023               MI              CT               ML Engineer   
2          2023               MI              CT               ML Engineer   
3          2023               SE              FT            Data Scientist   
4          2023               SE              FT            Data Scientist   
...         ...              ...             ...                       ...   
3750       2020               SE              FT            Data Scientist   
3751       2021               MI              FT  Principal Data Scientist   
3752       2020               EN              FT            Data Scientist   
3753       2020               EN              CT     Business Data Analyst   
3754       2021               SE              FT      Data Science Manager   

       salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0       80000             EUR          85847                 ES           100   
1       30000             USD          30000                 US           100   
2       25500             USD          25500                 US           100   
3      175000             USD         175000                 CA           100   
4      120000             USD         120000                 CA           100   
...       ...             ...            ...                ...           ...   
3750   412000             USD         412000                 US           100   
3751   151000             USD         151000                 US           100   
3752   105000             USD         105000                 US           100   
3753   100000             USD         100000                 US           100   
3754  7000000             INR          94665                 IN            50   

     company_location company_size  
0                  ES            L  
1                  US            S  
2                  US            S  
3                  CA            M  
4                  CA            M  
...               ...          ...  
3750               US            L  
3751               US            L  
3752               US            S  
3753               US            L  
3754               IN            L  

[3755 rows x 11 columns]

In [3]:
columns = df[['work_year', 'experience_level', 'job_title', 'salary_in_usd', 'employee_residence', 'company_location', 'company_size']]

columns.head(len(columns))

work_year experience_level                 job_title  salary_in_usd  \
0          2023               SE  Principal Data Scientist          85847   
1          2023               MI               ML Engineer          30000   
2          2023               MI               ML Engineer          25500   
3          2023               SE            Data Scientist         175000   
4          2023               SE            Data Scientist         120000   
...         ...              ...                       ...            ...   
3750       2020               SE            Data Scientist         412000   
3751       2021               MI  Principal Data Scientist         151000   
3752       2020               EN            Data Scientist         105000   
3753       2020               EN     Business Data Analyst         100000   
3754       2021               SE      Data Science Manager          94665   

     employee_residence company_location company_size  
0                    ES               ES            L  
1                    US               US            S  
2                    US               US            S  
3                    CA               CA            M  
4                    CA               CA            M  
...                 ...              ...          ...  
3750                 US               US            L  
3751                 US               US            L  
3752                 US               US            S  
3753                 US               US            L  
3754                 IN               IN            L  

[3755 rows x 7 columns]

I have selected a few columns that caught my attention.

The dataset includes several positions within the field of data analysis. I have decided to focus on the 'Data Analyst' position.

In [4]:
data_analyst = columns[columns['job_title'] == 'Data Analyst']

data_analyst.shape

(612, 7)

The dataset contains 612 rows related to the 'Data Analyst' position.

In [5]:
years = data_analyst.groupby('work_year')

years_dict = {years: group for years, group in years}

year_20 = years_dict.get(2020)
year_21 = years_dict.get(2021)
year_22 = years_dict.get(2022)
year_23 = years_dict.get(2023)

I have created a dictionary with the values corresponding to each year in the dataset.

After cleaning and preparing the dataset, the first objective of my analysis will be to determine the average salary for each experience level per year.

In [6]:
years = ['year_20', 'year_21', 'year_22', 'year_23']

result = {}

for year in years:
    entry = globals()[year][globals()[year]['experience_level'] == 'EN']
    entry_calc = entry['salary_in_usd'].sum() / len(entry)
    result[year] = entry_calc

result_df = pd.DataFrame(result, index=[0])

result_df.head()


year_20       year_21       year_22     year_23
0  40376.2  55926.285714  49971.894737  69523.1875

The average annual salary for an entry-level data analyst is as follows:

1.   2020 - 40,376.20 in USD
2.   2021 - 55,926.28 in USD
3.   2022 - 49,971.89 in USD
4.   2023 - 69,523.18 in USD

In [7]:
years = ['year_20', 'year_21', 'year_22', 'year_23']

result = {}

for year in years:
    entry = globals()[year][globals()[year]['experience_level'] == 'MI']
    entry_calc = entry['salary_in_usd'].sum() / len(entry)
    result[year] = entry_calc

result_df = pd.DataFrame(result, index=[0])

result_df.head()

year_20    year_21      year_22        year_23
0  46586.333333  75427.875  105486.3875  102252.407895

The average annual salary for a mid-level data analyst is as follows:

1.   2020 - 46,586.33 in USD
2.   2021 - 75,427.87 in USD
3.   2022 - 105,486.38 in USD
4.   2023 - 102,252.40 in USD

In [9]:
years = ['year_20', 'year_21', 'year_22', 'year_23']

result = {}

for year in years:
    entry = globals()[year][globals()[year]['experience_level'] == 'SE']
    entry_calc = entry['salary_in_usd'].sum() / len(entry)
    result[year] = entry_calc

result_df = pd.DataFrame(result, index=[0])

result_df.head()

<ipython-input-9-0a9c5b9584b5>:7: RuntimeWarning: invalid value encountered in long_scalars
  entry_calc = entry['salary_in_usd'].sum() / len(entry)


year_20  year_21        year_22        year_23
0      NaN  96769.5  114062.085714  125788.944724

The average annual salary for a senior-level data analyst is as follows:

1.   2020 - The dataset does not include senior-level data for the year 2020.
2.   2021 - 96,769.5 in USD
3.   2022 - 114,062.08 in USD
4.   2023 - 125,788.94 in USD

Next objective will be to get the average salary for the four year period.